src/middlewared/middlewared/plugins/webdav.py

```diff
@@ -114,9 +114,9 @@ class Config:
 
             await self._service_change('webdav', 'reload')
 
-        if data['perm']:
+        if new['perm']:
```

Comment:
Isnt this going to apply permission on every update? Even if perm was not changed?